agent/engine/task_manager.go

```diff
@@ -539,7 +539,8 @@ func (mtask *managedTask) isContainerFound(container *apicontainer.Container) bo
 }
 
 func (mtask *managedTask) isResourceFound(res taskresource.TaskResource) bool {
-	for _, r := range mtask.Resources {
+	resources := mtask.GetResources()
+	for _, r := range resources {
```

Comment:
nit: same comment as above. can we put these on a single line? same comment for instances of this below.